blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
+            n_dartboard.append(new_coord)
+            dartboard.append(old_coord)
 
-        probabilities: list of floats, default=None
-            A list of n probabilities,
-            where probabilities[i] corresponds to the probaility of moves[i]
-            being selected to perform its associated move() method.
+        self.n_dartboard = n_dartboard
+        self.dartboard = dartboard
 
-            If None, uniform probabilities are assigned.
+    def move(self, nc_context):
+        """
+        Function for performing smart darting move with darts that
+        depend on particle positions in the system
         """
 
-    #make a list from moves if not a list
-        if isinstance(moves,list):
-            self.moves = moves
-        else:
-            self.moves = [moves]
-        #normalize probabilities
-        if probabilities is None:
-            single_prob = 1. / len(self.moves)
-            self.probs = [single_prob for x in (self.moves)]
-        else:
-            prob_sum = float(sum(probabilities))
-            self.probs = [x/prob_sum for x in probabilities]
-        #if move and probabilitiy lists are different lengths throw error
-        if len(self.moves) != len(self.probs):
-            print('moves and probability list lengths need to match')
-            raise IndexError
-
-
-    def runEngine(self, context):
-        """Selects a random Move object based on its
-        assigned probability and and performs its move() function
-        on a context.
-        Parameters
-        ----------
-        context : openmm.context object
-        OpenMM context whose positions should be moved.
-        """
-        rand_num = np.random.choice(len(self.probs), p=self.probs)
-        try:
-            new_context = self.moves[rand_num].move(context)
-        except Exception as e:
-            #In case the move isn't properly implemented, print out useful info
-            print('Error: move not implemented correctly, printing traceback:')
-            ex_type, ex, tb = sys.exc_info()
-            traceback.print_tb(tb)
-            print(e)
-            raise SystemExit
-
-        return new_context
+        atom_indices = self.atom_indices
+        context = nc_context
+
+        stateinfo = context.getState(True, True, False, True, True, False)
+        oldDartPos = stateinfo.getPositions(asNumpy=True)
+        lig_pos = np.asarray(oldDartPos._value)[self.atom_indices]*unit.nanometers
+        self._findDart(context)
+        center = self.getCenterOfMass(lig_pos, self.masses)
+        selectedboard, changevec = self._calc_from_center(com=center)
+        if selectedboard != None:
+            #TODO just use oldDartPos instead of using temp newDartPos
+            newDartPos = np.copy(oldDartPos)
+            comMove = self._reDart(changevec)
+            vecMove = comMove - center
+            for residue in atom_indices:
```

Comment:
Um, wouldn't this be "for atomnr in atom_indices"? Or, these aren't atom_indices... ?